ballerina/tests/test.bal

```diff
@@ -0,0 +1,260 @@
+// under the License.
+
+import ballerina/test;
+
+// Configurable variables for environment setup.
+configurable boolean isLiveServer = ?;
+configurable string token = ?;
+configurable string serviceUrl = isLiveServer ? "https://api.openai.com/v1" : "http://localhost:9090";
+configurable string apiKey = isLiveServer ? token : "";
+
+// Initialize the connection configuration and client.
+final ConnectionConfig config = {auth: {token: apiKey}};
+final Client openaiFinetunes = check new Client(config, serviceUrl);
+
+// Define sample file content and name.
+final string fileName = "sample.jsonl";
+const byte[] fileContent = [123,13,10,32,32,32,32,34,112,114,111,109,112,116,34,58,32,34,87,104,97,116,32,105,115,32,116,104,101,32,97,110,115,119,101,114,32,116,111,32,50,43,50,34,44,13,10,32,32,32,32,34,99,111,109,112,108,101,116,105,111,110,34,58,32,34,52,34,13,10,125];
+
+// Record type to hold test data.
+public type TestData record {
+    string fileId;
+    string modelId;
+    string jobId;
+};
+
+// Initialize test data.
+TestData testData = {fileId: "", modelId: "", jobId: ""};
+
+// Function to generate test data.
+function dataGen() returns TestData[][] {
+    return [[testData]];
+}
+
+// Test case to list models.
+@test:Config {
+    dataProvider:  dataGen
+}
+isolated function testListModels(TestData testData) returns error? {
+
+    ListModelsResponse modelsResponse = check openaiFinetunes->/models.get();
+
+    testData.modelId = "gpt-3.5-turbo";
+
+    test:assertEquals(modelsResponse.'object, "list", "Object type mismatched");
+    test:assertTrue(modelsResponse.hasKey("data"), "Response does not have the key 'data'");
+}
+
+// Test case to retrieve a specific model by ID.
+@test:Config {
+    dataProvider:  dataGen,
+    dependsOn: [testListModels]
+}
+isolated function testRetrieveModel(TestData testData) returns error? {
+
+    string modelId = testData.modelId;
+    Model modelResponse = check openaiFinetunes->/models/[modelId].get();
+
+    test:assertEquals(modelResponse.id, modelId, "Model id mismatched");
+    test:assertTrue(modelResponse.hasKey("object"), "Response does not have the key 'object'");
+}
+
+// Test case to delete a model.
+@test:Config {
+    dependsOn: [testCreateFineTuningJob, testListModels, testRetrieveModel, testListFineTuningJobCheckpoints, testListFineTuningEvents],
+    dataProvider:  dataGen,
+    enable: isLiveServer? false : true // Enable this test only for mock server.
+}
+isolated function testDeleteModel(TestData testData) returns error? {
+
+    string modelIdCreated = testData.modelId;
+    DeleteModelResponse modelResponseDelete = check openaiFinetunes->/models/[modelIdCreated].delete();
+
+    test:assertEquals(modelResponseDelete.id, modelIdCreated, "Model id mismatched");
+    test:assertTrue(modelResponseDelete.hasKey("object"), "Response does not have the key 'object'");
+}
+
+// Files
+// Test case to list all files.
+@test:Config {}
```

Comment:
I think rather having comments to group the tests, following is the preferred approach.
```suggestion
@test:Config {
    groups: ["Files"]
}
```